examples/equilibrium.py

```diff
@@ -55,13 +56,16 @@
 eqs # results is (for now) a list of equilibrium objects
 
 s = eqs[0][1].dr.endo_grid.nodes().ravel()
+i=0
 for (w,eq) in eqs:
     dens = eq.μ.sum(axis=0) # \mu is a 2d array: exogenous x endogenous states
     # we sum all agents of a given type across the exogenous dimension
     plt.plot(s, dens, label=f"β={dist[i][1]['β']:.3f}")
+    i+=1
 plt.grid()
 
-# Oups, the more patient agents are diverging...
+# Looked like the patient agents were diverging but that was because the k grid was not large enough.
+# They are rich, but looks like not diverging.  (This reminds me: What is assumed about the distribution of assets beyond the upper cutoff?)  If we can recover the equilibrium interest rate, we can be sure: without growth, the impatience condition is R β < 1
```

Comment:
Here, there is no assumption about the distribution. All the mass ends up in the last point.
I'm wondering how information about the constraint could be coded up: here beta*R<1, tells us that it should work in theory, but it could also provide information to the optimizer about the range where to restrict the search for a solution.

Reply:
On the $R \beta < 1$ point, much of my [BufferStockTheory](http://econ.jhu.edu/people/ccarroll/papers/BufferStockTheory) paper is about deriving the most general results I can that define when the problem has a solution.  The last line of the [BufferStockTheory REMARK](https://mybinder.org/v2/gh/llorracc/BufferStockTheory/master?filepath=Code%2FPython%2FBufferStockTheory.ipynb)) is:  

[Two tables in the paper](https://econ.jhu.edu/people/ccarroll/papers/BufferStockTheory/#Factors-Defined-And-Compared) summarize the various definitions, and then articulate conditions required for the problem to have a nondegenerate solution.  Among the nondegenerate cases, the most interesting result is that if the Growth Impatience Condition holds there will be a target level of wealth.

And, there is a fairly comprehensive optional reporting system.  If you set the "verbose=true" flag in the ".checkConditions()" module, you get all sorts of info about whether the parameter values you have fed in are ones where there is a solution.  

In other papers where I have estimated the time preference rate, I've used the conditions to put bounds on the parameter search, which yielded a major improvement in the robustness and speed of finding the stable parameter values.  Basically, you can compute the "edge" value of $\bar{\beta}$ such that the problem diverges if $\beta \geq \bar{\beta}$.  The you can define, say, $\hat{\beta} = exp(-\nu) + \bar{\beta}$ and search for $\nu$, knowing that even as $\nu \uparrow infty$ you'll never actually propose to the algorithm a $\hat{\beta}$ for which the model is degenerate.